intelmq/bots/collectors/kafka/collector.py

```diff
@@ -0,0 +1,45 @@
+"""
+Kafka Collector Bot
+
+Collects information from the Apache Kafka distributed stream processing system.
+
+Args:
+    topic (str): topic to collect information from
+    bootstrap_servers (str): the ‘host[:port]’ string of the Apache Kafka system. Defaults to `localhost:9092`
+"""
+
+from intelmq.lib.bot import CollectorBot
+from intelmq.lib.exceptions import MissingDependencyError
+
+try:
+    import kafka
+except ImportError:
+    kafka = None
+
+class KafkaCollectorBot(CollectorBot):
+
+    def init(self):
+        if kafka is None:
+            raise MissingDependencyError("kafka")
+
+        self.topic = []
+        if getattr(self.parameters, 'topic', '') != '':
+            self.topic = self.parameters.topic
+
+        self.bootstrap_servers = 'localhost:9092'
+        if getattr(self.parameters, 'bootstrap_servers', '') != '':
+            self.bootstrap_servers = self.parameters.bootstrap_servers
+
+        self.logger.debug(f"topic set to {self.topic}, bootstrap_servers set to {self.bootstrap_servers}")
+
+    def process(self):
+        " Fetch messages from the Kafka server and pass them on one by one "
+        consumer = kafka.KafkaConsumer(self.topic, bootstrap_servers=self.bootstrap_servers, auto_offset_reset='earliest', consumer_timeout_ms=100, group_id=self._Bot__bot_id)
```

Comment:
Here we could use some generic parameters:
- `ssl_client_certificate` as `ssl_certfile`
- `ssl_ca_certificate` as `ssl_cafile`
- `http_verify_cert` as `ssl_check_hostname`
- _maybe_ `http_timeout_sec` as `request_timeout_ms`, `session_timeout_ms`, `heartbeat_interval_ms` and `consumer_timeout_ms`

Maybe we need to add authentication as well, but later

Reply:
I did that, but one thing: the doc says about `http_verify_cert`: `path to trusted CA bundle or directory, false to ignore verifying SSL certificates, or true (default) to verify SSL certificates` -  but`ssl_check_hostname`only allows `true` or `false`. Maybe we should not use the generic variable for the kafka collector?

Reply:
you're right, that can cause conflicts and confusion

Reply:
I have now changed the variable to `ssl_check_hostname`. I think it confusing to have a variable with `http` in the name used for something that has nothing to do with http.